terraform/gcp/cloud_build_deploy.tf

```diff
@@ -0,0 +1,114 @@
+
+  role    = "roles/container.developer"
+  members = [
+    "serviceAccount:${google_service_account.juniper_cloudbuild_service_account.email}"
+  ]
+}
+
+resource "google_cloudbuild_worker_pool" "juniper-deployment-worker-pool" {
+  name     = "juniper-deployment-worker-pool"
+  project  = var.project
+  location = var.region
+  worker_config {
+    disk_size_gb   = 100
+    machine_type   = "e2-medium"
+    no_external_ip = true
+  }
+  network_config {
+    peered_network = google_compute_network.cloud_build_network.id
+  }
+  depends_on = [google_service_networking_connection.private_vpc_connection]
+}
+
+resource "google_cloudbuild_trigger" "dev_auto_deploy_on_tag_push" {
+  name = "deploy-juniper"
+
+  location = "us-central1"
+
+  github {
+    owner = "broadinstitute"
+    name = "juniper"
+    push {
+      tag = "[0-9]+.[0-9]+.[0-9]+"
+    }
+  }
+
+  build {
+
+    step {
+      name = "gcr.io/cloud-builders/git"
+      id = "Clone repository"
+      args = ["clone", "--depth", "1", "--branch", "$TAG_NAME", "https://github.com/broadinstitute/juniper.git", "/workspace/juniper" ]
+    }
+
+    step {
+      name = "gcr.io/cloud-builders/docker"
+      id = "Wait for images to be published"
+      entrypoint = "bash"
+      args = [
+        "-c",
+        <<EOT
+function wait_for_image () {
+    echo "attempting to pull $1..."
+    until docker pull $1
+    do
+        echo "image $1 not found..."
+        sleep 30
+        echo "attempting to pull $1..."
+    done
+    echo "found $1!"
+}
+
+wait_for_image "us-central1-docker.pkg.dev/broad-juniper-eng-infra/juniper/juniper-admin:$TAG_NAME"
+wait_for_image "us-central1-docker.pkg.dev/broad-juniper-eng-infra/juniper/juniper-participant:$TAG_NAME"
+        EOT
+      ]
+    }
+
+    step {
+      name = "gcr.io/google.com/cloudsdktool/cloud-sdk"
+      id = "Deploy to k8s"
+      dir = "/workspace/juniper/terraform/gcp/k8s"
+      entrypoint = "bash"
+      env = ["CLOUDSDK_COMPUTE_ZONE=us-central1", "CLOUDSDK_CONTAINER_CLUSTER=juniper-cluster"]
+        args = [
+            "-c",
+            <<EOT
+gcloud container clusters get-credentials juniper-cluster --zone us-central1
+curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash
+helm upgrade -f environments/$_ENV.yaml juniper . --namespace $_NAMESPACE --set appVersion=$TAG_NAME --atomic --wait --timeout 15m
+            EOT
+        ]
+    }
+
+    options {
+      logging = "CLOUD_LOGGING_ONLY"
+      worker_pool = google_cloudbuild_worker_pool.juniper-deployment-worker-pool.id
+    }
+
+    timeout = "1500s" # 25 minutes so that the images have time to be built and pushed
+  }
+
+  service_account = "projects/${var.project}/serviceAccounts/${google_service_account.juniper_cloudbuild_service_account.email}"
+
+  substitutions = {
+    _ENV = var.environment
+    _NAMESPACE = var.k8s_namespace
+  }
+
+  approval_config {
+    approval_required = var.environment == "prod" ? true : false
+  }
```

Comment:
What I wanted to do was create a manual trigger that could be applied to any tag that matches our semvar tags. However, if you create a manual trigger, it cannot accept regex for the tag name, it can only be a static tag. So I settled for the in-between solution of creating an automatic trigger that requires approval. So, every time we push up a PR, it's gonna create a new job but not actually execute it. We can either accept the job we want to run or manually execute on the tag we want.